### ISSUE TITLE
Inconsistent installation script: ruby-dev, nodejs should not be installed before checking if already exists

### ISSUE THREAD

**Issue by imtiazShakil** [beefproject/beef#1911]
Verify first that your issue/request has not been posted previously:
yes, a similar issue #1704 exists but this one is slightly different.

#### Environment

What version/revision of BeEF are you using? v0.5.00 (master)
On what version of Ruby? 2.5.3
On what browser? chrome
On what operating system? Lubuntu 19.03

#### Summary
In wiki, you've written a detailed approach of installing `rvm` to manage `ruby`. But at the end of the day, when user is installing beef via script `./install` it installs `ruby-dev` which installs system wide ruby. The whole point of `rvm` was to manage ruby at a user-level (or system level depending on user). The same goes for `nodejs` also.

#### Expected Behaviour
Before installing `ruby-dev`, `nodejs` installation script must check if they already exists. 

#### Actual Behaviour
`install` script installs `ruby-dev`, `nodejs` from linux repository without checking if they are already managed/exist.

**Comment by bcoles** [comment]
Yeah it's a mess. You're correct that the install process needs some work.

There's more context in #1477, specifically comments https://github.com/beefproject/beef/issues/1477#issuecomment-364824370 and https://github.com/beefproject/beef/issues/1477#issuecomment-371816205, concluding:

> Adding in proper detection for the environment, and offering the choice of installing system ruby, or rbvenv, or rvm, will take some effort. I opted for the simplest method: allowing the user to set up their own Ruby environment, rather than force installing RVM and Ruby and gems which is rather impolite.

**Comment by DeezyE** [comment]
@bcoles I'm picking this one up. I've looked over 1477 for context. Is the linked comment from 1477 still valid?
Basically trying to work out which direction you recommend moving forward on this - should we be ignoring ruby package managers?

**Comment by bcoles** [comment]
> 
> 
> @bcoles I'm picking this one up. I've looked over 1477 for context. Is the linked comment from 1477 still valid?

Some of the comments in #1477 are still valid. I have no idea regarding the status of the installer for Mac.

> Basically trying to work out which direction you recommend moving forward on this - should we be ignoring ruby package managers?

Whoever is maintaining BeEF will have to decide.

For Ruby, ideally, the installer should detect whether rbenv or rvm are present and use them - optionally prompting the user which to use if both are present, then fall back to using system Ruby. I see no reason why the installer should install rvm or rbenv if not present.

For NodeJS, the installer should detect whether nodejs is already installed, in which case, presumably the best course of action is to do nothing.

**Comment by DeezyE** [comment]
Thanks

**Comment by bcoles** [comment]
The Ruby version management should be resolved in #1940

The NodeJS management is yet to be resolved.
